docs(): update middlewares for fastify (#2589)
nestjs/docs.nestjs.com: content/techniques/performance.md

new FastifyAdapter({ logger: true });
```

#### Middleware

Middleware functions retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page](https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,

```typescript
@@filename(logger.middleware)
import { Injectable, NestMiddleware } from '@nestjs/common';
import { FastifyRequest, FastifyReply } from 'fastify';

@Injectable()
export class LoggerMiddleware implements NestMiddleware {
  use(req: FastifyRequest['raw'], res: FastifyReply['raw'], next: () => void) {
    console.log('Request...');
    next();
  }
}
@@switch
import { Injectable } from '@nestjs/common';

@Injectable()
export class LoggerMiddleware {
  use(req, res, next) {
    console.log('Request...');
    next();
  }
}
```

> 

Reviewer: ```suggestion
```

new FastifyAdapter({ logger: true });
```

#### Middleware

Middleware functions retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page](https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,

> 

Reviewer: ```suggestion
```

new FastifyAdapter({ logger: true });
```

#### Middleware

> The middleware with `fastify` "works", but it retrieves only the raw `req` and `res` objects and not the fastify wrapper. This is a side effect of how `middie` works and `fastify` already talks about it in its documents for more information see [here](https://www.fastify.io/docs/latest/Reference/Middleware/).

Reviewer: ```suggestion
Middleware functions (when `@nestjs/platform-fastify` is used) retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page(https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,
```

> Middleware functions (when `@nestjs/platform-fastify` is used) retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page(https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,

Reviewer: ```suggestion
Middleware functions (when `@nestjs/platform-fastify` is used) retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page](https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,
```

> Middleware functions (when `@nestjs/platform-fastify` is used) retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page](https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,

Reviewer: ```suggestion
Middleware functions retrieve the raw `req` and `res` objects instead of Fastify's wrappers. This is how the `middie` package works (that's used under the hood) and `fastify` - check out this [page](https://www.fastify.io/docs/latest/Reference/Middleware/) for more information,
```

> > warning **Warning** The middleware in `fastify` has an IncomingMessage request type `FastifyRequest['raw']` and a ServerResponse type `FastifyReply['raw']`.

Reviewer: ```suggestion
> warning **Warning** The middleware in `fastify` has an IncomingMessage request type `FastifyRequest['raw']` and a ServerResponse type `FastifyReply['raw']`.
```

cc @jmcdo29 any ideas on how to rewrite it to make it sound more native-ish 🤔

Reviewer 2: ```suggestion
> warning **Warning** When using `fastify` as the HTTP engine the request parameter of middleware has a type of `FastifyRequest['raw']` and the response parameter has a type `FastifyReply['raw']`.
```

> > warning **Warning** When using `fastify` as the HTTP engine the request parameter of middleware has a type of `FastifyRequest['raw']` and the response parameter has a type `FastifyReply['raw']`.

Reviewer: ```suggestion
```
